google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/SpannerPool.java

```diff
@@ -155,10 +156,17 @@ static class SpannerPoolKey {
 
     @VisibleForTesting
     static SpannerPoolKey of(ConnectionOptions options) {
```

Comment:
```suggestion
    static SpannerPoolKey of(ConnectionOptions options) throws SpannerException {
```

Reply:
I don't agree with that. This exception falls in the following category "__Runtime exceptions represent problems that are the result of a programming problem, and as such, the API client code cannot reasonably be expected to recover from them or to handle them in any way.__" There's nothing reasonable that we can do to catch and handle this exception in the client library, and it should naturally bubble up to the client application.